### ISSUE TITLE
Option to not include sources in output

### ISSUE THREAD

**Issue by Fishrock123** [eslint/eslint#11170]
**The version of ESLint you are using.**
5.9.0

**The problem you want to solve.**
Eslint output is unnecessarily large, in particular when using child processes to parallelize eslint.

**Your take on the correct solution to problem.**
Add an option to not include the whole sources in output (json output).

**Are you willing to submit a pull request to implement this change?**
Yeah I'd be willing to do this.

**Comment by g-plane** [comment]
You can try `eslint . --formatter=json`.

**Comment by g-plane** [comment]
However, there is no way to pass options to formatter. Maybe you can write a custom one?

**Comment by Fishrock123** [comment]
Ah, I didn't know custom formatters could be written so that they are loaded directly.

I solved this by writing the following custom formatter:
```js
'use strict'

module.exports = function(results) {
  // Delete the full sources off the object...
  for (const result of results) {
    delete result.source
  }
  return JSON.stringify(results)
}
```

Thanks @g-plane
